app_users/models.py

```diff
@@ -233,6 +234,16 @@ def get_or_create_personal_workspace(self) -> tuple["Workspace", bool]:
 
         return Workspace.objects.get_or_create_from_user(self)
 
+    @cached_property
+    def cached_workspaces(self) -> list["Workspace"]:
+        from workspaces.models import Workspace
+
+        return list(
+            Workspace.objects.filter(
+                memberships__user=self, memberships__deleted__isnull=True
+            ).order_by("-is_personal", "-created_at")
+        ) or [self.get_or_create_personal_workspace()[0]]
```

Comment:
more efficient would be to make use of `self.memberships`, and use `AppUser.objects.prefetch_related("memberships__workspace")` when retrieving the user.

we also fetch the memberships to figure out the role in some cases. this can be `[m.workspace for m in self.memberships]`. this would be only 2 DB queries - one for appuser, and one for membership-workspace - throughout the request-response cycle. other than the runs.

ref: https://docs.djangoproject.com/en/5.1/ref/models/querysets/#prefetch-related

Reply:
makes sense, lets do that too